src/canvas/heatmapFeature.js

```diff
@@ -124,54 +124,23 @@ var canvas_heatmapFeature = function (arg) {
    * @protected
    */
   ////////////////////////////////////////////////////////////////////////////
-  this._colorize = function (context2d, width, height, imageData, gradient) {
-    var isLittleEndian = true, i, j, index, pixels;
-
-    // Determine whether Uint32 is little- or big-endian.
-    if (!m_typedBuffer || (m_typedBuffer.length !== imageData.data.length)) {
-      m_typedBuffer = new ArrayBuffer(imageData.data.length);
+  this._colorize = function (pixels, gradient) {
+    var grad = new Uint32Array(gradient.buffer);
+    var i, j, k, pixlen = pixels.length;
```

Comment:
I'll make it so there is just one statement.

I'd be shocked if there is a measurable performance difference between single and multiple var statements.